### PR TITLE
JDBC Homework

### PR DESCRIPTION
**Pull request by IvanKhomka**
Added solution for jdbc homework

### CODE REVIEW

File: src/main/java/jdbctask/Demonstration.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import jdbctask.dao.UserDao;
+import jdbctask.oldprojectclasses.Ticket;
+import jdbctask.oldprojectclasses.User;
+
+import java.util.List;
+
+public class Demonstration {
+    private static final String url = "jdbc:postgresql://localhost:5432/my_ticket_service_db";
+    private static final String username = "postgres";
+    private static final String password = "82463755";
```
Comment (Alexmansar) [inline src/main/java/jdbctask/Demonstration.java:13]: if we use private static final feilds - we shoul give name like URL, USERNAME...

File: src/main/java/jdbctask/dao/TicketDao.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        this.username = username;
+        this.password = password;
+    }
+
+    private Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(url, username, password);
+    }
+
+    public void saveTicket(int userId, String ticketType) {
+        String query = "INSERT INTO Ticket (user_id, ticket_type) VALUES (?, ?::ticket_type);";
```
Comment (Alexmansar) [inline src/main/java/jdbctask/dao/TicketDao.java:24]: use it like constants . In each query 
```
 private static final String url = "jdbc:postgresql://localhost:5432/my_ticket_service_db";
    private static final String username = "postgres";
    private static final String password = "82463755";
```

File: src/main/java/jdbctask/oldprojectclasses/Ticket.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    public int getId() {
+        return id; }
+    public int getUserId() {
+        return userId; }
+    public String getTicketType() {
+        return ticketType; }
+    public LocalDateTime getCreationDate() {
+        return creationDate; }
+}
```
Comment (Alexmansar) [inline src/main/java/jdbctask/oldprojectclasses/Ticket.java:27]: fix formatting

File: src/main/java/jdbctask/oldprojectclasses/User.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        this.creationDate = creationDate;
+    }
+
+
+    public int getId() {
+        return id; }
+    public String getName() {
+        return name; }
+    public LocalDateTime getCreationDate() {
+        return creationDate; }
```
Comment (Alexmansar) [inline src/main/java/jdbctask/oldprojectclasses/User.java:22]: here

File: src/main/java/jdbctask/dao/UserDao.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    public UserDao(String dbUrl, String username, String password) {
+        this.url = dbUrl;
+        this.username = username;
+        this.password = password;
+    }
+
+    private Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(url, username, password);
+    }
```
Comment (Alexmansar) [inline src/main/java/jdbctask/dao/UserDao.java:19]: please, create new class that should work with connection, and just return preperad one
alo, you can remove 
```
 private final String url;
    private final String username;
    private final String password;
```
and use it in new class

and 
```
 private static final String url = "jdbc:postgresql://localhost:5432/my_ticket_service_db";
    private static final String username = "postgres";
    private static final String password = "82463755";
```
as well should be use in new class
its good way to have separate class for this connection logic

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@IvanKhomka Hi, In general good job, but I have some general comments. Please, fix it)

File: src/main/java/jdbctask/DBConfig.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package jdbctask;
+
+public class DBConfig {
+    public static final String DB_URL = "jdbc:postgresql://localhost:5432/my_ticket_service_db";
+    public static final String USERNAME = "postgres";
+    public static final String PASSWORD = "82463755";
```
Comment (Alexmansar) [inline src/main/java/jdbctask/DBConfig.java:6]: move it just to DBConnection

File: src/main/java/jdbctask/DBConnection.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    private Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(url, username, password);
+    }
+
+    public PreparedStatement createPreparedStatement(String sql) throws SQLException {
+        Connection connection = getConnection();
+        return connection.prepareStatement(sql);
+    }
```
Comment (Alexmansar) [inline src/main/java/jdbctask/DBConnection.java:26]: move before private method

File: src/main/java/jdbctask/dao/TicketDao.java (hunk 2 above)
Comment (Alexmansar) [reply src/main/java/jdbctask/dao/TicketDao.java:19]: what about it? If its not clear - just ask)

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@IvanKhomka Hi)) I have another commets)) please have a look)

File: src/main/java/jdbctask/dao/TicketDao.java (hunk 2 above)
Comment (Alexmansar) [reply src/main/java/jdbctask/dao/TicketDao.java:18]: String query = "INSERT INTO Ticket (user_id, ticket_type) VALUES (?, ?::ticket_type);";

use as a constant in the class))

File: src/main/java/jdbctask/DBConfig.java (hunk 6 above)
Comment (Alexmansar) [reply src/main/java/jdbctask/DBConfig.java]: And this toy can set in the constructor here DBConnection

File: src/main/java/jdbctask/dao/TicketDao.java (hunk 2 above)
Comment (IvanKhomka) [reply src/main/java/jdbctask/dao/TicketDao.java:18]: Ok, sorry, just did't see this)

**Review by Alexmansar — APPROVED** [review]
@IvanKhomka Could be merged))

**Pull request merged**